Objects/unicodeobject.c

```diff
@@ -14617,6 +14617,60 @@ unicode_new_impl(PyTypeObject *type, PyObject *x, const char *encoding,
     return unicode;
 }
 
+static const char *
+as_const_char(PyObject *obj, const char *name)
+{
+    if (!PyUnicode_Check(obj)) {
+        PyErr_Format(PyExc_TypeError,
+                     "str() argument '%s' must be str, not %T",
+                     name, obj);
+        return NULL;
+    }
+    const char *str = _PyUnicode_AsUTF8NoNUL(obj);
+    if (str == NULL) {
+        return NULL;
+    }
+    return str;
+}
+
```

Comment:
You may add a comment to mention that this function is a fast-path for positional-only cases.